NachoSpigot-Server/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java

```diff
@@ -533,7 +533,8 @@ private static EntityDamageEvent handleEntityDamageEvent(Entity entity, DamageSo
         } else if (source == DamageSource.FALL) {
             cause = DamageCause.FALL;
         } else if (source == DamageSource.GENERIC) {
-            return new EntityDamageEvent(entity.getBukkitEntity(), null, modifiers, modifierFunctions);
+            //return new EntityDamageEvent(entity.getBukkitEntity(), null, modifiers, modifierFunctions);
+            cause = DamageCause.CUSTOM; // KigPaper
```

Comment:
This will most likely break some plugins, though I'm unsure

Reply:
@LewUwU your opinion? otherwise i revert it

Reply:
I don't really know, i prefer to leave this like the original code to prevent issues